cf_units/__init__.py

```diff
@@ -753,7 +770,50 @@ def num2date(time_value, unit, calendar):
     if unit_string.endswith(" since epoch"):
         unit_string = unit_string.replace("epoch", EPOCH)
     cdftime = netcdftime.utime(unit_string, calendar=calendar)
-    return cdftime.num2date(time_value)
+    return _num2date_to_nearest_second(time_value, cdftime)
+
+
+def _num2date_to_nearest_second(time_value, utime):
+    # Return datetime encoding of numeric time value with respect to the given
+    # time reference units, with a resolution of 1 second.
+    try:
```

Comment:
I prefer the pattern used above: testing for __iter__ to this try except pattern